release-notifier/internal/webscrape/webscrape.go

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package webscrape
+
+import (
+	"fmt"
+	"github.com/gocolly/colly"
+	"regexp"
+)
+
+// Semantic Versioning schema regex
+const regexPattern = `^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`
+
+func GetLatestPSQLRelease() string {
```

Comment:
Is there any long term experience, if this is a reliable way to retrieve the version. 
I mean it seems to heavily rely on the HTML structure, that we do not control

Reply:
That's good question and I don't have a good answer for that since I didn't find a better way of extracting new release version than using webscraping. If you know other sources/api, let me know pls.

Reply:
Hmm then I guess let's try with that and get our own experience